plugins/inputs/systemd_units/systemd_units_linux.go

```diff
@@ -314,12 +341,23 @@ func (s *SystemdUnits) Gather(acc telegraf.Accumulator) error {
 			continue
 		}
 
+		var tags map[string]string
 		// Create the metric
-		tags := map[string]string{
-			"name":   state.Name,
-			"load":   state.LoadState,
-			"active": state.ActiveState,
-			"sub":    state.SubState,
+		if s.scope == "user" {
+			tags = map[string]string{
+				"name":   state.Name,
+				"load":   state.LoadState,
+				"active": state.ActiveState,
+				"sub":    state.SubState,
+				"user":   s.user,
+			}
+		} else {
+			tags = map[string]string{
+				"name":   state.Name,
+				"load":   state.LoadState,
+				"active": state.ActiveState,
+				"sub":    state.SubState,
+			}
 		}
```

Comment:
I think this is more readable
```suggestion
		tags := map[string]string{
			"name":   state.Name,
			"load":   state.LoadState,
			"active": state.ActiveState,
			"sub":    state.SubState,
		}
		if s.scope == "user" {
			tags["user"] =  s.user
		}
```